src/unstated.js

```diff
@@ -119,23 +119,17 @@ export type ProviderProps = {
 };
 
 export function Provider(props: ProviderProps) {
-  return (
-    <StateContext.Consumer>
```

Comment:
you've removed really important feature of `unstated` here - nested `Provider`s, you must revert this